tests/transform_dialect/cuda/attention.mlir

```diff
@@ -0,0 +1,114 @@
+func.func @attention(%query: tensor<192x1024x64xf32>, %key: tensor<192x1024x64xf32>, %value: tensor<192x1024x64xf32>) -> tensor<192x1024x64xf32> {
```

Comment:
it's not ideal to have such lit test in this folder as this only runs when there is a GPU target and that makes maintaining those tests painful. What part are you trying to test here? If it is the codegen pipeline for attention I would suggest moving this in LLVMGPU codegen otherwise break it up into smaller tests for individual transformations.

Reply:
I was initially thinking of moving this to iree/samples but given that the test is closely related to the code, do you think it is fine to keep it in LLVMGPU/Codegen?